apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -374,7 +421,7 @@ class PageTemplateModal extends Component {
 									) }
 							</form>
 							<TemplateSelectorPreview
-								blocks={ this.getBlocksByTemplateSlug( previewedTemplate ) }
+								blocks={ modifyParsedBlocks( this.getBlocksByTemplateSlug( previewedTemplate ) ) }
```

Comment:
This is where we modify the blocks that are used only in the preview. Any modified blocks are returned as new block references via `cloneBlock` else the blocks are modified everywhere due to references to the original `blocks` object being used everywhere!

Reply:
I can see a potential bottleneck here, as this will be called on every single render. 

I'm all for running this once and memoize it in some way. If we don't want to update the blocks that will be inserted into the document on template selection, let's make a new object to store block specifically for previews that will have this processing applied.

Does it make sense?

Reply:
Yep. As in lets have one referenced to the modified blocks and one ref to the original blocks rather than recomputing all the time.

Reply:
I've fixed this and memoized the results.